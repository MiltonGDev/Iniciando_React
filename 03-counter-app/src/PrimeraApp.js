import React, { Fragment } from 'react';
import PropTypes from 'prop-types';


//asi es como creamos props, como argumentos recibidos en PrimeraApp, que es nuestro componente
const PrimeraApp = ( 
    {
        saludo,
        subtitulo
    } 
     ) => {

    
    
    // console.log(saludo)


    return (
        <>

            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>
        
        </>

        ); 

} 

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subtitulo: 'soy un subtitulo por defecto 2'
}

export default PrimeraApp;