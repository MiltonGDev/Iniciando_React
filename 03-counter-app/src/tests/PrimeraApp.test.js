import React from "react";
import PrimeraApp from "../PrimeraApp";
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('Pruebas en PrimeraApp', () => {
    
    // test('debe de mostrar el mensaje "Hola, soy Milton" ', () => {
    //     const saludo = 'Hola, soy Milton';

    //     const { getByText } = render( <PrimeraApp saludo={saludo}/> );

    //     expect( getByText(saludo) ).toBeInTheDocument();    
    // })


    test('Debe de momstrar <PrimeraApp /> Correctamente', () => {

        const saludo = 'Hola, soy Milton';


        const wrapper = shallow( <PrimeraApp saludo = {saludo}/> );

        expect( wrapper ).toMatchSnapshot();


    })

    test('Debe de mostrar el subtitulo enviado por props', () => {
        
        const saludo = 'Hola, soy Milton';
        const subtitulo = 'este es el nuevo subtitulo';


        const wrapper = shallow( <PrimeraApp saludo = {saludo} subtitulo={subtitulo}/> );

        const textoParrafo = wrapper.find('p').text();
        // console.log(textoParrafo);

        expect( textoParrafo ).toBe( subtitulo );




    })
    
    
    

})
