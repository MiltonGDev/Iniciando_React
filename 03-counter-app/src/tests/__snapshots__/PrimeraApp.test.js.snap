// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe de momstrar <PrimeraApp /> Correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Milton
  </h1>
  <p>
    soy un subtitulo por defecto 2
  </p>
</Fragment>
`;
